hoomd/md/pytest/test_update_group_dof.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import hoomd
+import pytest
+
+
+@pytest.fixture(scope='session')
+def snapshot_factory(device):
+    """Make a test snapshot for update_group_dof testing."""
+
+    def make_snapshot():
+        """Make the snapshot."""
+        snap = hoomd.Snapshot(device.communicator)
+
+        if snap.communicator.rank == 0:
+            box = [10, 10, 10, 0, 0, 0]
+            snap.configuration.box = box
+            snap.particles.N = 3
+            snap.particles.position[:] = [[0, 1, 0], [-1, 1, 0], [1, 1, 0]]
+            snap.particles.velocity[:] = [[0, 0, 0], [0, -1, 0], [0, 1, 0]]
+            snap.particles.moment_inertia[:] = [
+                [2.0, 0, 0],
+                [1, 1, 1],
+                [1, 1, 1],
+            ]
+            snap.particles.angmom[:] = [[0, 2, 4, 6]] * 3
+            snap.particles.types = ['A', 'B']
```

Comment:
Why set multiple particle types if no particles are of the `B` type?

Reply:
`test_filter_updater` uses the B type and a type filter to change the number of degrees of freedom with a `FilterUpdater`. The other tests don't use it, but it takes a few characters less code to provide it in the snapshot fixture regardless.